Team9Project/src/com/indragie/cmput301as1/BaseTabFragment.java

```diff
@@ -0,0 +1,292 @@
+	 * Index of a item that is long pressed.
+	 */
+	private int longPressedItemIndex;
+	
+	/**
+	 * Active user.
+	 */
+	private User user;
+
+	/** An array of items to display in ArrayList */
+    
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+    	
+    	setUserFromPreferences();
+    	setHasOptionsMenu(true);
+        
+    	
+    	load();
+		
+    	/** Creating array adapter to set data in listview */
+        ExpenseClaimArrayAdapter adapter = new ExpenseClaimArrayAdapter(getActivity(), listModel.getItems());
+ 
+        /** Setting the array adapter to the listview */
+        setListAdapter(adapter);
+        
+        
+		
+ 
+        return super.onCreateView(inflater, container, savedInstanceState);
+    }
+    
+    @Override
+    public void onActivityCreated(Bundle savedInstanceState){
+    	super.onActivityCreated(savedInstanceState);
+    	
+    	
+    	final ActionMode.Callback longClickCallback = new ActionMode.Callback() {
+			@Override
+			public boolean onActionItemClicked(ActionMode mode, MenuItem item) {
+				switch (item.getItemId()) {
+				case R.id.action_delete:
+					startDeleteAlertDialog();
+					mode.finish();
+					return true;
+				default:
+					return false;
+				}
+			}
+
+			@Override
+			public boolean onCreateActionMode(ActionMode mode, Menu menu) {
+				mode.getMenuInflater().inflate(R.menu.contextual_delete, menu);
+				return true;
+			}
+
+			@Override
+			public void onDestroyActionMode(ActionMode mode) {}
+
+			@Override
+			public boolean onPrepareActionMode(ActionMode mode, Menu menu) {
+				return false;
+			}
+		};
+    	getListView().setOnItemLongClickListener(new OnItemLongClickListener() {
+			@Override
+			public boolean onItemLongClick(AdapterView<?> parent, View view, int position, long id) {
+				longPressedItemIndex = position;
+				getActivity().startActionMode(longClickCallback);
+				return true;
+			}
+		});
+    }
+    
+    @Override
+	public void onDestroy() {
+		listModel.deleteObserver(this);
+		super.onDestroy();
+	}
+    
+    @SuppressWarnings("static-access")
+	@Override
+	public void onActivityResult(int requestCode, int resultCode, Intent data) {
+		if (resultCode != getActivity().RESULT_OK) return;
+		switch (requestCode) {
+		case ADD_EXPENSE_CLAIM_REQUEST:
+			onAddExpenseResult(data);
+			break;
+		case EDIT_EXPENSE_CLAIM_REQUEST:
+			onEditExpenseResult(data);
+			break;
+		case SORT_EXPENSE_CLAIM_REQUEST:
+			onSortExpenseResult(data);
+			break;
+		}
+	}
+    
+    /**
+	 * Changes the sorting mode based on a comparator chosen by {@link ExpenseClaimSortActivity}
+	 * @param data The intent to get the comparator from.
+	 */
+	@SuppressWarnings("unchecked")
+	private void onSortExpenseResult(Intent data) {
+		Comparator<ExpenseClaim> comparator = (Comparator<ExpenseClaim>)data.getSerializableExtra(ExpenseClaimSortActivity.EXPENSE_CLAIM_SORT);
+		listModel.setComparator(comparator);
+		refresh();
```

Comment:
It's because you called addObserver() on the activity instead of the fragment (I mentioned this in another comment). Once you fix that it will work.



--
Indragie Karunaratne

On Thu, Apr 2, 2015 at 8:22 PM, Nic Carroll <notifications@github.com>
wrote:

>> +			break;
>> +		case SORT_EXPENSE_CLAIM_REQUEST:
>> +			onSortExpenseResult(data);
>> +			break;
>> +		}
>> +	}
>> +    
>> +    /**
>> +	 * Changes the sorting mode based on a comparator chosen by {@link ExpenseClaimSortActivity}
>> +	 * @param data The intent to get the comparator from.
>> +	 */
>> +	@SuppressWarnings("unchecked")
>> +	private void onSortExpenseResult(Intent data) {
>> +		Comparator<ExpenseClaim> comparator = (Comparator<ExpenseClaim>)data.getSerializableExtra(ExpenseClaimSortActivity.EXPENSE_CLAIM_SORT);
>> +		listModel.setComparator(comparator);
>> +		refresh();
> I've tried this implementation, but it seems that update is not called on changes.
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/CMPUT301W15T09/Team9Project/pull/33/files#r27712573